### PR TITLE
1366 csv improvements

### PR DESCRIPTION
**Pull request by willdoran**
This pull request makes the following changes:
- Added import validator to disregard task requirements for published posts during import

Test these changes by:
-

Fixes ushahidi/platform# .

Ping @ushahidi/platform

---
This change is [Reviewable](https://reviewable.io/reviews/ushahidi/platform/1405)

### CODE REVIEW

File: application/classes/Ushahidi/Core.php [hunk 1]
```diff
@@ -511,6 +511,19 @@ public static function init()
 			'post_value_factory' => $di->lazyGet('repository.post_value_factory'),
 			'post_value_validator_factory' => $di->lazyGet('validator.post.value_factory'),
 			];
+
+			$di->params['Ushahidi_Validator_Post_Import'] = [
```
Comment (rjmackay) [inline application/classes/Ushahidi/Core.php]: I think this is unnecessary. AuraDI can figure out the params needed from the parent class

File: application/classes/Ushahidi/Validator/Form/Create.php [hunk 2]
```diff
@@ -18,8 +18,8 @@ class Ushahidi_Validator_Form_Create extends Ushahidi_Validator_Form_Update
 	protected function getRules()
 	{
 		return array_merge_recursive(parent::getRules(), [
-			'name' => [['not_empty'],
-			[[$this, 'checkPostTypeLimit'], [':validation']],
+			'values' => [
```
Comment (rjmackay) [inline application/classes/Ushahidi/Validator/Form/Create.php]: .. I'm not sure why we're changing this at all?

File: application/classes/Ushahidi/Validator/Post/Import.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+/**
+ * Ushahidi Post Create Validator
+ *
+ * @author     Ushahidi Team <team@ushahidi.com>
+ * @package    Ushahidi\Application
+ * @copyright  2014 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+use Ushahidi\Core\Entity;
```
Comment (rjmackay) [inline application/classes/Ushahidi/Validator/Post/Import.php]: Can we drop all the `use` statements? we don't use them here, only in the parent

File: application/classes/Ushahidi/Validator/Post/Import.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+use Ushahidi\Core\Traits\AdminAccess;
+use Ushahidi\Core\Traits\Permissions\ManagePosts;
+use Ushahidi\Core\Usecase\Post\UpdatePostRepository;
+use Ushahidi\Core\Usecase\Post\UpdatePostTagRepository;
+
+class Ushahidi_Validator_Post_Import extends Ushahidi_Validator_Post_Create
+{
+  protected function getRules()
+	{
+		return array_merge(parent::getRules(), [
```
Comment (rjmackay) [inline application/classes/Ushahidi/Validator/Post/Import.php]: Needs a comment to explain whats changing and why

File: application/classes/Ushahidi/Validator/Post/Import.php (hunk 3 above)
Comment (willdoran) [inline application/classes/Ushahidi/Validator/Post/Import.php]: yep

File: application/classes/Ushahidi/Validator/Form/Create.php (hunk 2 above)
Comment (willdoran) [inline application/classes/Ushahidi/Validator/Form/Create.php]: That's a mistake I'll revert it

File: application/classes/Ushahidi/Validator/Post/Import.php (hunk 4 above)
Comment (willdoran) [inline application/classes/Ushahidi/Validator/Post/Import.php]: done

File: application/classes/Ushahidi/Core.php (hunk 1 above)
Comment (willdoran) [inline application/classes/Ushahidi/Core.php]: done

**Comment by willdoran** [comment]
I'll add a unit test now

**Pull request merged**